feat: edit mode preference

fuscodev opened the pull request:
Allow users with edit role to lock/unlock pages (using _is_locked_ column in _pages_ table that is still unused)

#399 

![readonly](https://github.com/user-attachments/assets/cd484766-749e-41c2-b141-45906491f3c6)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/docmost/docmost?pullRequest=666) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/docmost/docmost?pullRequest=666) before we can accept your contribution.<br/><hr/>**fuscodev** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/docmost/docmost?pullRequest=666) it.</sub>

Philipinho commented:
Hi @fuscodev, this looks good.
I am thinking about having  a user preference for page edit state. I think that would make page lock in its current form redundant.

My Idea for it is this. If a user has their default page edit preference set to readonly, there would be a [segmented control](https://mantine.dev/core/segmented-control/) component on each page which they can toggle. By default, each page they access will be in "Reading state" unless they click on the edit option (or shortcut e key).
<img width="173" alt="image" src="https://github.com/user-attachments/assets/ba1f5022-617d-4f99-a21c-0c9c27c172cf" />

I can find another use for the `isLocked` column in the future.

Is this something you would be willing to work on? I appreciate your contributions to the editor.
Thank you.

fuscodev commented:
Hi @Philipinho, in the last commit I removed the use of the `isLocked` column and added the page edit state preference.

- default: edit
- the switch in the page view doesn't change the default preference (instead of `fullWidth`)

![readonly](https://github.com/user-attachments/assets/cd484766-749e-41c2-b141-45906491f3c6)

edersong commented:
Does this feature have a release date?
I'm asking because it seems to be complete about two months.

MilenFrom commented:
Hi, any news on this?

fuscodev commented:
> I running docmost as a docker service with version v0.20.4,but I cannot find the reading/edit button that you have mentioned。Please tell me the way  to get that?

It's a PR, only @Philipinho can merge it

Philipinho commented:
The current blocker for this is me trying to figure out the best UI for this. I don’t want the editor’s top bar to become too crowded.
I am thinking we could put the toggle in the editor menu item like we did with “Full width”.

Ideas are welcome.

MilenFrom commented:
Guys, the UpNote app has really nice UI for this feature and it is very convenient - at least for me.
You can see the screenshot where the button is - I'll share it here.
So basically how this is made is that this button is a toggle switching the document between two modes:
- Editing mode
- Reading mode

Can be a good idea if per workspace the user to be able to se a default mode for each new document.

![SCR-20250604-kqly](https://github.com/user-attachments/assets/c4f7f5ae-e11f-41a9-9ff3-6b8ea022b07e)

jackeyt commented:
> Guys, the UpNote app has really nice UI for this feature and it is very convenient - at least for me. You can see the screenshot where the button is - I'll share it here. So basically how this is made is that this button is a toggle switching the document between two modes:
> 
> * Editing mode
> * Reading mode
> 
> Can be a good idea if per workspace the user to be able to se a default mode for each new document.
> 
> ![SCR-20250604-kqly](https://private-user-images.githubusercontent.com/14107346/451287669-c4f7f5ae-e11f-41a9-9ff3-6b8ea022b07e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDkxMDg2NTAsIm5iZiI6MTc0OTEwODM1MCwicGF0aCI6Ii8xNDEwNzM0Ni80NTEyODc2NjktYzRmN2Y1YWUtZTExZi00MWE5LTlmZjMtNmI4ZWEwMjJiMDdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjA1VDA3MjU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiYzRiNjQzYWFjZjUxNTUwOWQzYWI0ZDM3YTg4NDU2M2Q4NjFiODgyZjU2OWNmMzE0NjY1MDliY2FiNTgzOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3R02qaYIfL_S4uM2Rw2D52-mYCaj-6CKZFUX9M8vXuU)

So when can I be able to get this feature?

Hellgringo83 commented:
Having this toggle button would be a great improvement. For my use case most users only read and some edit. So normal behaviour would be the locked mode. 
I know that this was not the initial idea so it could also be an option to set the default behaviour for an instance or even only enable the "lock" feature by a server parameter.

fuscodev commented:
It's ok for me, but the menu button and breadcumbs won't appear on small devices if we use segmented control

![Screenshot 2025-06-12 002116](https://github.com/user-attachments/assets/1980c3a7-6a54-4cbb-8b9f-2ef8e5f5e57e)
